flytekit/image_spec/image_spec.py

```diff
@@ -278,6 +278,16 @@ def exist(self) -> Optional[bool]:
                     f"    pip install --upgrade docker"
                 )
 
+            if isinstance(e, DockerException) and "Error while fetching server API version" in str(e):
+                click.secho(
```

Comment:
If the image exists in your `ghcr.io` registry and docker is not running, then that should still work. This message tells a user to install Docker, but it is not strictly required.

Reply:
Yup. But I don't think we want to hide the fact that docker should generally be running. If it's not clear, the user will eventually run into images that don't exist in their registry and they will get image pull failures they don't understand. This happened today.

Reply:
maybe I should change the "error message" wording in the PR title, but this change doesn't block images being found and used if the docker is off but the image exists

Reply:
Let's write out these details in the message. Something like: "Docker is not running and flytekit will assume that the image exists in your registry. If you want to build the image, please install docker: ..."

Reply:
So I was going to change the `click.secho(f"Flytekit assumes the image {img_name} already exists.", fg="blue")` in `should_build()`, but that could occur even if docker is running, for example an `APIError` that returns `None`. 

The current suggestion actually mentions that docker isn't running if it catches the exact error docker throws, then the `Flytekit assumes the image ghcr.io/dansola/test-image:ORI1BZVtVKVludLUJsq_Rg already exists.` log still comes up after. 

Do you think it's not clear that those two are associated? Happy to keep it general and just toss in the mention of docker in `should_build()`, but I just don't want to sound that alarm if it isn't the case.